### PR TITLE
[Serve][Docs] SkyServe On-demand Spot Doc. 

### PR DESCRIPTION
**Pull request by MaoZiming**
https://github.com/skypilot-org/skypilot/issues/3440

Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `conda deactivate; bash -i tests/backward_compatibility_tests.sh`

### CODE REVIEW

File: docs/source/serving/spot-policy.rst [hunk 1]
```diff
@@ -0,0 +1,140 @@
+.. _spot_policy:
+
+Using Spot Instances
```
Comment (romilbhardwaj) [inline docs/source/serving/spot-policy.rst:3]: nit - to keep toc unambiguous:
```suggestion
Using Spot Instances for Serving
```

File: docs/source/serving/spot-policy.rst [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+    resources:
+        ports: 8081
+        cpus: 2+
+        use_spot: true
+
+    workdir: examples/serve/http_server
+
+    run: python3 server.py
+
```
Comment (romilbhardwaj) [inline docs/source/serving/spot-policy.rst:28]: Can we add a tip/note for Kubernetes, noting that Kubernetes pods are counted as on-demand replicas and that this policy can do a mix of Kubernetes + cloud spot?

File: docs/source/serving/spot-policy.rst [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+.. code-block:: yaml
+
+    service:
+        readiness_probe: /health
+        replica_policy:
+            min_replicas: 2
+            max_replicas: 3
+            target_qps_per_replica: 1
+            base_ondemand_fallback_replicas: 1
+
```
Comment (romilbhardwaj) [inline docs/source/serving/spot-policy.rst:19]: Good to explain the YAML inline:
```suggestion
            min_replicas: 2
            max_replicas: 3
            target_qps_per_replica: 1
            # Ensures that one of the replicas is run on on-demand instances
            base_ondemand_fallback_replicas: 1

```

File: docs/source/serving/spot-policy.rst [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        use_spot: true
+
+    workdir: examples/serve/http_server
+
+    run: python3 server.py
+
+Dynamic On-demand Fallback
+==========================
+
+SkyServe supports dynamically fallback to on-demand replicas when spot replicas are not available. This is enabled by setting :code:`dynamic_ondemand_fallback` to be :code:`true`. This is useful for ensuring the required capacity of replicas in the case of spot instance interruptions. When spot replicas are available, SkyServe will automatically switch back to using spot replicas to maximize cost savings. SkyServe supports both specifying :code:`base_ondemand_fallback_replicas` and :code:`dynamic_ondemand_fallback`.
```
Comment (romilbhardwaj) [inline docs/source/serving/spot-policy.rst:32]: Maybe we should put `SkyServe supports both specifying :code:base_ondemand_fallback_replicas and :code:dynamic_ondemand_fallback.` in a tip/hint box for visibility?

File: docs/source/serving/spot-policy.rst [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+.. code-block:: yaml
+
+    service:
+        readiness_probe: /health
+        replica_policy:
+            min_replicas: 2
+            max_replicas: 3
+            target_qps_per_replica: 1
+            dynamic_ondemand_fallback: true
+
```
Comment (romilbhardwaj) [inline docs/source/serving/spot-policy.rst:43]:
```suggestion
            target_qps_per_replica: 1
            # Allows replicas to be run on on-demand instances if spot instances are not available
            dynamic_ondemand_fallback: true

```

**Review by romilbhardwaj — COMMENTED** [review]
Thanks @MaoZiming! Took a quick look.

**Comment by romilbhardwaj** [comment]
Pushed a few minor fixes and addressed comments above.

**Review by romilbhardwaj — APPROVED** [review]

**Review by Michaelvll — APPROVED** [review]

**Pull request merged**